lib/filters/hostname.ts

```diff
@@ -1,7 +1,11 @@
-export const HostnameFilter = function () {
-  var refreshFunctions = [];
-  var timer;
-  var input = document.createElement("input");
+import { Node } from "../utils/node";
+import { CanRender } from "../container";
+import { Filter } from "../datadistributor";
+
+export const HostnameFilter = function (): CanRender & Filter {
+  let refreshFunctions: (() => any)[] = [];
+  let timer: NodeJS.Timeout;
```

Comment:
Technically `NodeJS` doesn't exist in the browser.  
You can use this to get the correct type in any case:

```suggestion
  let timer: ReturnType<typeof setTimeout>;
```